src/Api/MediaLibraryApi.php

```diff
@@ -160,14 +160,102 @@ public function mediaPackagePackageNameGet(string $package_name, ?int $limit = 2
    */
   public function mediaFileIdGet(int $id, &$responseCode, array &$responseHeaders)
   {
-    $responseCode = Response::HTTP_I_AM_A_TEAPOT;
+    $json_response_array = [];
+    $media_package_file = $this->entity_manager->getRepository(MediaPackageFile::class)
+      ->findOneBy(['id' => $id])
+        ;
+
+    if (null !== $media_package_file)
+    {
+      $responseCode = Response::HTTP_OK; // 200 => OK
+      $json_response_array[] = new MediaFile($this->getMediaFileResponseData($media_package_file));
+    }
+    else
+    {
+      $responseCode = Response::HTTP_NOT_FOUND; // => Not found
+
+      return null;
+    }
+    $responseData = new MediaFiles(['media_files' => $json_response_array]);
+
+    return $responseData;
   }
 
   /**
    * {@inheritdoc}
    */
   public function mediaFilesGet(?int $limit = 20, ?int $offset = 0, ?string $flavor = null, &$responseCode, array &$responseHeaders)
   {
-    $responseCode = Response::HTTP_I_AM_A_TEAPOT;
+    if (null === $limit)
+    {
+      $limit = 20;
+    }
+    if (null === $offset)
+    {
+      $offset = 0;
+    }
+
+    $total_results = 0;
+
+    $json_response_array = [];
+    $media_package_files = null;
+    if (null == $flavor)
+    {
+      $query = $this->entity_manager->createQuery('SELECT u FROM App\Entity\MediaPackageFile u');
+      $media_package_files = $query->getResult();
+
+      $num_query = $this->entity_manager->createQuery('SELECT COUNT(u.id) FROM App\Entity\MediaPackageFile u');
+      $total_results = $num_query->getSingleScalarResult();
+    }
+    else
+    {
+      $query = $this->entity_manager->createQuery('SELECT u FROM App\Entity\MediaPackageFile u WHERE u.flavor = :flavor')
```

Comment:
If you go if Option 2:

This flavor condition could be dynamically added to the query only if needed, heavily reducing the duplicated code